dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp

```diff
@@ -0,0 +1,166 @@
+                                     .setHighVoltageLimit(420.0)
+                                     .add();
+
+  powsybl::iidm::Bus& vl3Bus1 = vl3.getBusBreakerView().newBus().setId("VL3_BUS1").add();
+
+  powsybl::iidm::VoltageLevel& vl4 = substation2.newVoltageLevel()
+                                     .setId("VL4")
+                                     .setName("VL4_NAME")
+                                     .setTopologyKind(powsybl::iidm::TopologyKind::BUS_BREAKER)
+                                     .setNominalVoltage(225.0)
+                                     .setLowVoltageLimit(200.0)
+                                     .setHighVoltageLimit(260.0)
+                                     .add();
+
+  vl4.getBusBreakerView().newBus().setId("VL4_BUS1").add();
+
+  powsybl::iidm::Line& MyLine = network.newLine()
+                                    .setId("VL1_VL3")
+                                    .setVoltageLevel1(vl1.getId())
+                                    .setBus1(vl1Bus1.getId())
+                                    .setConnectableBus1(vl1Bus1.getId())
+                                    .setVoltageLevel2(vl3.getId())
+                                    .setBus2(vl3Bus1.getId())
+                                    .setConnectableBus2(vl3Bus1.getId())
+                                    .setR(3.0)
+                                    .setX(33.33)
+                                    .setG1(1.0)
+                                    .setB1(0.2)
+                                    .setG2(2.0)
+                                    .setB2(0.4)
+                                    .add();
+
+  LineInterfaceIIDM li(MyLine);
+  ASSERT_EQ(li.getID(), "VL1_VL3");
+  ASSERT_EQ(li.getVNom1(), 380.0);
+  ASSERT_EQ(li.getVNom2(), 360.0);
+  ASSERT_EQ(li.getR(), 3.0);
+  ASSERT_EQ(li.getX(), 33.33);
+  ASSERT_EQ(li.getG1(), 1.0);
+  ASSERT_EQ(li.getG2(), 2.0);
+  ASSERT_EQ(li.getB1(), 0.2);
+  ASSERT_EQ(li.getB2(), 0.4);
+
+  ASSERT_EQ(li.getBusInterface1().get(), nullptr);
+  ASSERT_EQ(li.getBusInterface2().get(), nullptr);
+  const boost::shared_ptr<BusInterface> x_b1(new BusInterfaceIIDM(vl1Bus1));
+  const boost::shared_ptr<BusInterface> x_b3(new BusInterfaceIIDM(vl3Bus1));
+  li.setBusInterface1(x_b1);
+  li.setBusInterface2(x_b3);
+  ASSERT_EQ(li.getBusInterface1().get()->getID(), "VL1_BUS1");
+  ASSERT_EQ(li.getBusInterface2().get()->getID(), "VL3_BUS1");
+
+  MyLine.getTerminal1().disconnect();
+  ASSERT_EQ(li.getVNom1(), 0.0);
+  MyLine.getTerminal2().disconnect();
+  ASSERT_EQ(li.getVNom2(), 0.0);
+  MyLine.getTerminal1().connect();
+  ASSERT_EQ(li.getVNom1(), 380.0);
+  MyLine.getTerminal2().connect();
+  ASSERT_EQ(li.getVNom2(), 360.0);
+
+  ASSERT_TRUE(li.getInitialConnected1());
+  ASSERT_TRUE(li.getInitialConnected2());
+
+  // FAIRE: ici manquent qqs VoltageLevelInterface
+  // FAIRE: ici manquent qqs CurrentLimitInterface
+
+  ASSERT_EQ(li.getP1(), 0.0);
```

Comment:
The case in which the terminal1 or 2 of the line has a P or a Q defined is not tested.
We should have 4 tests here: getInitialConnected=True/False and getQ is a NaN or is defined

Reply:
I don't know yet how to set InitialConnected to false. In fact that is not tested since there lacks a 'else something' in the code. That makes we fall in default case, which is the same as NaN. That may explain why LCOV don't mention something to test here. I do one little push to show that point. coverage raises up to 73% now.

Reply:
getInitialConnected to false is easy: you just call MyLine.getTerminal1().disconnect() and then li.getInitialConnected1() (and it should be the first time you call it).
I am looking on how we can set values for the P/Q of the terminals

Reply:
OK, I see that you found the way to init P/Q, so now the only missing test is one for which we disconnect both terminal before calling getInitialConnect and then getP1 and others should always return 0

Reply:
All is done now :)